OpenFGA v1.3.8 panic

Our CI just caught what looks like a regression in v1.3.8 as it was passing just minutes prior to the new release appearing on Github.

Stack trace looks like:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xf6ad52]

goroutine 576 [running]:
github.com/openfga/openfga/pkg/server/commands.(*WriteCommand).Execute(0x1498c08?, {0x1487640?, 0xc00059fdd0?}, 0x2?)
	/home/runner/work/openfga/openfga/pkg/server/commands/write.go:68 +0x32
github.com/openfga/openfga/pkg/server.(*Server).Write(0xc0002ea000, {0x1487640?, 0xc00059fd70?}, 0xc0005934a0)
	/home/runner/work/openfga/openfga/pkg/server/server.go:527 +0x5cd
github.com/openfga/api/proto/openfga/v1._OpenFGAService_Write_Handler.func1({0x1487640, 0xc00059fd70}, {0x12284a0?, 0xc0005934a0})
	/home/runner/go/pkg/mod/github.com/openfga/api/proto@v0.0.0-20231201195548-7ad5c05e75ca/openfga/v1/openfga_service_grpc.pb.go:371 +0x75
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.UnaryServerInterceptor.func34({0x1487640, 0xc00059fb00}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0, 0xc0004d29a8)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.1/interceptors/auth/auth.go:48 +0xb0
google.golang.org/grpc.getChainUnaryHandler.func1({0x1487640, 0xc00059fb00}, {0x12284a0, 0xc0005934a0})
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1163 +0xb2
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.NewLoggingInterceptor.UnaryServerInterceptor.func71({0x1487640, 0xc00059fb00}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0?, 0xc00022ae00)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.1/interceptors/server.go:22 +0x28a
google.golang.org/grpc.getChainUnaryHandler.func1({0x1487640, 0xc00059fb00}, {0x12284a0, 0xc0005934a0})
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1163 +0xb2
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.NewUnaryInterceptor.UnaryServerInterceptor.func69({0x1487640, 0xc00059fad0}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0?, 0xc00022ad00)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.1/interceptors/server.go:22 +0x28a
google.golang.org/grpc.getChainUnaryHandler.func1({0x1487640, 0xc00059fad0}, {0x12284a0, 0xc0005934a0})
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1163 +0xb2
github.com/grpc-ecosystem/go-grpc-middleware/tags.UnaryServerInterceptor.func1({0x1487640?, 0xc00059fa70?}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0, 0xc00022acc0)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/tags/interceptors.go:23 +0x9c
google.golang.org/grpc.getChainUnaryHandler.func1({0x1487640, 0xc00059fa70}, {0x12284a0, 0xc0005934a0})
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1163 +0xb2
github.com/openfga/openfga/pkg/middleware/validator.UnaryServerInterceptor.func1.1({0x1487640, 0xc00059f9e0}, {0x12284a0, 0xc0005934a0})
	/home/runner/work/openfga/openfga/pkg/middleware/validator/validator.go:32 +0x86
github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/validator.UnaryServerInterceptor.func1({0x1487640, 0xc00059f9e0}, {0x12284a0, 0xc0005934a0}, 0xc0007b56f0?, 0xc000596ce0)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.1/interceptors/validator/interceptors.go:23 +0x6d
github.com/openfga/openfga/pkg/middleware/validator.UnaryServerInterceptor.func1({0x1487640, 0xc00059f9e0}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0?, 0xc00022ac80)
	/home/runner/work/openfga/openfga/pkg/middleware/validator/validator.go:31 +0x98
google.golang.org/grpc.getChainUnaryHandler.func1({0x1487640, 0xc00059f9e0}, {0x12284a0, 0xc0005934a0})
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1163 +0xb2
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.NewUnaryInterceptor.UnaryServerInterceptor.func65({0x1487640, 0xc00059f9b0}, {0x12284a0, 0xc0005934a0}, 0xc0007aa3c0?, 0xc00022ac00)
	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.1/interceptors/server.go:22 +0x28a
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x1487640, 0xc00059f9b0}, {0x12284a0, 0xc0005934a0}, 0xc000ab69f8?, 0x10dc380?)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1154 +0x85
github.com/openfga/api/proto/openfga/v1._OpenFGAService_Write_Handler({0x125ca60?, 0xc0002ea000}, {0x1487640, 0xc00059f9b0}, 0xc0007ba200, 0xc00041d700)
	/home/runner/go/pkg/mod/github.com/openfga/api/proto@v0.0.0-20231201195548-7ad5c05e75ca/openfga/v1/openfga_service_grpc.pb.go:373 +0x135
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003de960, {0x1487640, 0xc00059f920}, {0x148f980, 0xc000500680}, 0xc0007d4480, 0xc000318120, 0x1e2def8, 0x0)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1343 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0003de960, {0x148f980, 0xc000500680}, 0xc0007d4480)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1737 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:986 +0x86
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 45
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:997 +0x145
```

The query which caused this was:
```
+ timeout --foreground 120 /home/stgraber/data/code/go/bin/incus storage delete test-pool --verbose
ERROR  [2023-12-05T02:01:44Z] stgraber: writes=&[] deletions=&[{Object:storage_pool:test-pool Relation:server User:server:incus}] 
ERROR  [2023-12-05T02:01:44Z] Failed to remove storage pool from authorizer  error="Failed to delete tuple from OpenFGA store (user: \"server:incus\"; relation: \"server\"; object: \"storage_pool:test-pool\"): Post \"http://172.17.0.111:51589/stores/01HGVVBCJR9QWJSR6SZ83PATB8/write\": EOF" name=test-pool
```

This appears to hit the first time we send a write request with a deletion but no writes through the API endpoint.

Looking at our testsuite, all requests up till that point had something in writes.

--- a/pkg/server/commands/write.go
+++ b/pkg/server/commands/write.go
@@ -65,9 +65,11 @@ func NewWriteCommand(datastore storage.OpenFGADatastore, opts ...WriteCommandOpt
 func (c *WriteCommand) Execute(ctx context.Context, req *openfgav1.WriteRequest) (*openfgav1.WriteResponse, error) {
 	if !c.enableConditions {
 		tks := req.GetWrites()
-		for _, tk := range tks.TupleKeys {
-			if tk.Condition != nil {
-				return nil, status.Error(codes.InvalidArgument, "conditions not supported")
+		if tks != nil {
+			for _, tk := range tks.TupleKeys {
+				if tk.Condition != nil {
+					return nil, status.Error(codes.InvalidArgument, "conditions not supported")
+				}
 			}
 		}
 	}

